operator/pkg/resources/configuration/patch.go

```diff
@@ -145,6 +145,14 @@ func PropertiesEqual(
 	l logr.Logger, v1, v2 interface{}, metadata rpadmin.ConfigPropertyMetadata,
 ) bool {
 	log := l.WithName("PropertiesEqual")
+
+	// "null" (string) is the only way to define null as the desired value.
+	// Therefore, treat `"null"` as equal to `null` (only if `"null"` is on the k8s side)
+	if metadata.Nullable {
```

Comment:
So this fixes the diffing but are we going to have bugs with actually setting `null` values? If not does that mean we have duplicate code attempting to massage string values into the correct format? 😬

Reply:
we cannot set `null` because our type in the CRD is string, so it can never be an actual nil. well, at least have null on the CRD side of things. but i suspect there's some magic in redpanda to turn "null" into null - this stuff actually worked, "null" turned into null in redpanda.

definitely needs more test for sure, i will add them hopefully tomorrow